dao/src/main/java/org/thingsboard/server/dao/sql/cf/CalculatedFieldRepository.java

```diff
@@ -34,10 +37,16 @@ public interface CalculatedFieldRepository extends JpaRepository<CalculatedField
 
     Page<CalculatedFieldEntity> findAllByTenantIdAndEntityId(UUID tenantId, UUID entityId, Pageable pageable);
 
+    Page<CalculatedFieldEntity> findByTenantId(UUID tenantId, Pageable pageable);
+
     List<CalculatedFieldEntity> findAllByTenantId(UUID tenantId);
 
     List<CalculatedFieldEntity> removeAllByTenantIdAndEntityId(UUID tenantId, UUID entityId);
 
     long countByTenantIdAndEntityId(UUID tenantId, UUID entityId);
 
+    CalculatedFieldId findExternalIdById(UUID id);
```

Comment:
You don't even have external id in the CF! Don't implement any Exportable daos/services for calculated fields. Treat calculated field like a relation or attribute